src/test/java/CI/GitHandlerTest.java

```diff
@@ -7,6 +7,30 @@
 
 public class GitHandlerTest {
 
+  /**
+  * Test that a branch should run all steps when a push comes in.
+  * Runs a build on master which should always be successful
+  * Runs a build on a branch that fails on build
+  * Runs a build on a branch that fails to start the tests.
+  * Result is send to the email provided to event
+  * @author Andreas Gylling
+  */
+  @Test
+  public void testRequestPush(){
+    GitEvent event = new GitEvent("","");
+    // Master branch should always build sucessfully
+    event.setBranchName("master");
+    event.setPusherEmail("agylling@kth.se");
+    GitHandler gh = new GitHandler(event);
+    gh.request_push();
+    // Fail already at build.
+    event.setBranchName("nobuild");
+    gh.request_push();
+    // Fail at test after build
+    event.setBranchName("testsfail");
+    gh.request_push();
+  }
```

Comment:
We also need to check for a branch that doesn't exist. The current code should crash, I believe (because pull_branch will return a null).